device/src/drivers/ble/mesh/model/sensor/mod.rs

```diff
@@ -625,6 +630,26 @@ where
     pub fn new(data: C::Data<'a>) -> Self {
         Self { data }
     }
+
+    fn parse(parameters: &[u8]) -> Result<Self, ParseError> {
+        let mut pos = 0;
+        let mut result = None;
+        for d in C::DESCRIPTORS {
+            pos += 2;
+            let parameters = &parameters[pos..(pos + d.size)];
+
+            result = Some(C::Data::decode(d.id, parameters));
+        }
```

Comment:
Problem here is that parsing sensor data with more than 1 property becomes impossible. But with the proposed trait above, this code becomes:

```
let mut data = Default::default();
for d in C::DESCRIPTORS {
    // Decode property id and verify match
    let parameters = ...
    data.decode(id, parameters)?;
}